crates/astria-conductor/src/driver.rs

```diff
@@ -74,12 +82,22 @@ impl Driver {
         alert_tx: AlertSender,
     ) -> Result<(Self, executor::JoinHandle, Option<reader::JoinHandle>)> {
         let (cmd_tx, cmd_rx) = mpsc::unbounded_channel();
-        let (executor_join_handle, executor_tx) = executor::spawn(&conf, alert_tx.clone()).await?;
+        let (executor_join_handle, executor_tx) = executor::spawn(&conf, alert_tx.clone())
+            .await
+            .wrap_err("failed to construct block verifier")?;
+
+        let block_verifier = Arc::new(
+            BlockVerifier::new(&conf.tendermint_url)
+                .wrap_err("failed to construct BlockVerifier")?,
+        );
 
         let (reader_join_handle, reader_tx) = if conf.disable_finalization {
             (None, None)
         } else {
-            let (reader_join_handle, reader_tx) = reader::spawn(&conf, executor_tx.clone()).await?;
+            let (reader_join_handle, reader_tx) =
+                reader::spawn(&conf, executor_tx.clone(), block_verifier.clone())
+                    .await
+                    .wrap_err("failed to construct Reader")?;
```

Comment:
I know we refer to this as `reader`, but is there a better way to call this than `Reader` at least in the error? What is it reading?